src/modules/navigator/navigator_main.cpp

```diff
@@ -1043,10 +1061,21 @@ float Navigator::get_cruising_speed()
 	}
 }
 
-void Navigator::set_cruising_speed(float speed)
+void Navigator::set_cruising_speed(float speed, SpeedType type)
 {
+
 	if (_vstatus.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
-		_mission_cruising_speed_mc = speed;
+
+		//airspeed or groundspeed (does not differentiate currently)
+		if (type == SpeedType::AIR_SPEED || type == SpeedType::GROUND_SPEED) {
+			_mission_cruising_speed_mc = speed;
+
+		} else if (type == SpeedType::CLIMB_SPEED) {
+			_mission_vertical_up_speed_mc = speed;
+
+		} else if (type == SpeedType::DESCENT_SPEED) {
+			_mission_vertical_down_speed_mc = speed;
+		}
 
 	} else {
 		_mission_cruising_speed_fw = speed;
```

Comment:
So if you set `set_cruising_speed(2.f, SpeedType::CLIMB_SPEED)` while being a fixed-wing, you'll get the `_mission_cruising_speed_fw` set to 2?

Reply:
This would be correct as I have not implemented this feature for fixed wing as of now. 

@bresch however, I see what you are saying that a user might set CLIMB_SPEED for fixed wing, and have the vehicles cruising speed limited. Better to not do the operation than to have unexpected outcomes.

Would you suggest throwing the value away if anything other than AIR_SPEED or GROUND_SPEED is selected, or would we not want to merge in this enhancement without fixed wing support?

Reply:
> I see what you are saying that a user might set CLIMB_SPEED for fixed wing, and have the vehicles cruising speed limited.

I'm not sure how comfortable I am with a user selecting climb and descent speeds for a fixed-wing via command. They are highly vehicle specific. Should likely be left alone.

Reply:
Well they're as much vehicle specific as the horizontal airspeed setpoint, no? Speaking about FW_T_CLMB_R_SP and FW_T_SINK_R_SP.

Reply:
I am going to just make a commit that ignores the fixed wing climb and descent do_change_speed for now

Reply:
> Well they're as much vehicle specific as the horizontal airspeed setpoint, no? Speaking about FW_T_CLMB_R_SP and FW_T_SINK_R_SP.

Yes. But the two are coupled, and the climb rates are more constrained. Would rather have airspeed setpoint change coming in (which of course gets constrained) and let the controller handle how to deal with the climb and sink rates. not the other way around.

Otherwise.. if someone wants to continuously stream climb rate commands.. this is an entirely different API discussion.

> I am going to just make a commit that ignores the fixed wing climb and descent do_change_speed for now

sounds good.

Reply:
I tested this on a quad. Indeed there are some params needed to be set (they are changed in the new version indeed) as below. I just put all of them to 6 so that they can consume my desired velocities.
MPC Z VEL ALL = -6 
MPC_Z_ VEL_ MAX_DN = 6
MPC Z VEL MAX UP = 6
MPC Z V AUTO DN = 6
MPC_Z_V_AUTO_UP = 6

SITL Log: https://review.px4.io/plot_app?log=4b3eae0f-1bd9-469a-9268-4bcd3a026388

.plan file used (with slight changes) : https://drive.google.com/file/d/1YzKsnWMyO1FDTOismCYD9ytMS4xKIYe-/view

Pixhawk 6C Flight test:

https://review.px4.io/plot_app?log=a3fe0940-867a-4f54-83e6-94126a864ce8
https://review.px4.io/plot_app?log=52173558-4026-4ce5-8f8d-376b99f0f645
https://review.px4.io/plot_app?log=1a95b254-06e2-4500-a5fc-f0d93b632887

Let me know if there are any comments. 
cc: @afwilkin

Reply:
@bresch @tstastny 

Commit pushed to ignore fixed wing speed constraints in the vertical direction. Tested in SITL and posted log: https://review.px4.io/plot_app?log=e84e6f98-ce57-4522-9157-fc35f7fd2bcc

Reply:
commit here: https://github.com/PX4/PX4-Autopilot/pull/20495/commits/34f8e47300782c73bbfda5b86ce430ff014824e5